crates/core/tedge_actors/src/actors.rs

```diff
@@ -0,0 +1,304 @@
+use crate::ChannelError;
+use crate::ConcurrentServiceMessageBox;
+use crate::Message;
+use crate::MessageBox;
+use crate::ServiceMessageBox;
+use async_trait::async_trait;
+
+/// Enable a struct to be used as an actor.
+///
+///
+#[async_trait]
+pub trait Actor: 'static + Sized + Send + Sync {
+    /// Type of message box used by this actor
+    type MessageBox: MessageBox;
+
+    /// Return the actor instance name
+    fn name(&self) -> &str;
```

Comment:
I would really avoid to pass the actor name along to each and every `send` and `recv` method calls.

Currently, the actor name is given to its message box at build time, where it's use when logging the messages sent and received by the actor. I only wonder if that name is useful along the actor state itself. It can be known only by the actor builder and passed to the message box on build.

Reply:
The name of the actor is used for logging in 3 places:
* logging messages sent and received,
* logging key steps of the actor behavior,
* logging runtime events as actor spawned/cancelled/crash ...

Reply:
This will be addressed as a follow-up task.